src/webgpu/capability_info.ts

```diff
@@ -87,135 +87,1083 @@ export const kGeneratableErrorScopeFilters = kErrorScopeFilters.filter(
-  'depth32float':          [            ,              ,          ,        ,    true,     false,          ,      true,     false,              'depth',               4],
-  'depth16unorm':          [            ,              ,          ,        ,    true,     false,          ,      true,      true,              'depth',               2],
-  'stencil8':              [            ,              ,          ,        ,   false,      true,          ,      true,      true,               'uint',               1],
+/**
+ * Defaults applied to all tables automatically. Used only inside `tableWithDefaults`.
+ * This ensures keys are never missing, always explicitly `undefined`.
+ *
+ * All top-level keys must be defined here, or they won't be exposed.
+ */
+const kUniversalDefaults = {
+  blockWidth: undefined,
+  blockHeight: undefined,
+  color: undefined,
+  depth: undefined,
+  stencil: undefined,
+  colorRender: undefined,
+  multisample: undefined,
+  feature: undefined,
+  baseFormat: undefined,
+
+  sampleType: undefined,
+  copySrc: undefined,
+  copyDst: undefined,
+  bytesPerBlock: undefined,
+  renderable: false,
+  renderTargetPixelByteCost: undefined,
+  renderTargetComponentAlignment: undefined,
+
+  // IMPORTANT:
+  // Add new top-level keys both here and in TextureFormatInfo_TypeCheck.
+} as const;
+/**
+ * Takes `table` and applies `defaults` to every row, i.e. for each row,
+ * `{ ... kUniversalDefaults, ...defaults, ...row }`.
+ * This only operates at the first level; it doesn't support defaults in nested objects.
+ */
+function tableWithDefaults<Defaults extends {}, Table extends { readonly [K: string]: {} }>({
+  defaults,
+  table,
+}: {
+  defaults: Defaults;
+  table: Table;
+}): {
+  readonly [F in keyof Table]: {
+    readonly [K in keyof typeof kUniversalDefaults]: K extends keyof Table[F]
+      ? Table[F][K]
+      : K extends keyof Defaults
+      ? Defaults[K]
+      : typeof kUniversalDefaults[K];
+  };
+} {
+  return Object.fromEntries(
+    Object.entries(table).map(([k, row]) => [k, { ...kUniversalDefaults, ...defaults, ...row }])
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  ) as any;
+}
+
+/** "plain color formats", plus rgb9e5ufloat. */
+const kRegularTextureFormatInfo = tableWithDefaults({
+  defaults: { blockWidth: 1, blockHeight: 1 },
+  table: {
+    // plain, 8 bits per component
+
+    r8unorm: {
+      color: { type: 'float', copySrc: true, copyDst: true, storage: false, bytes: 1 },
+      colorRender: { blend: true, resolve: true, byteCost: 1, alignment: 1 },
+      renderable: true,
+      /*prettier-ignore*/ get renderTargetComponentAlignment() { return this.colorRender.alignment; },
+      /*prettier-ignore*/ get renderTargetPixelByteCost() { return this.colorRender.byteCost; },
+      multisample: true,
+      /*prettier-ignore*/ get sampleType() { return this.color.type; },
+      copySrc: true,
+      copyDst: true,
```

Comment:
I think for the regular formats `copySrc` and `copyDst` seem to always be `true` so maybe we can default them?